src/scirpy/ir_dist/metrics.py

```diff
@@ -392,6 +394,698 @@ def _compute_block(self, seqs1, seqs2, origin):
         return result
 
 
+class TCRdistDistanceCalculator:
+    """Computes pairwise distances between TCR CDR3 sequences based on the "tcrdist" distance metric.
+
+    The code of this class is heavily based on: https://github.com/agartland/pwseqdist/blob/master/pwseqdist
+
+    Using default weight, gap penalty, ntrim and ctrim is equivalent to the
+    original distance published in Dash et al, (2017).
+
+    Parameters
+    ----------
+    dist_weight : int
+        Weight applied to the mismatch distances before summing with the gap penalties
+    gap_penalty : int
+        Distance penalty for the difference in the length of the two sequences
+    ntrim/ctrim : int
+        Positions trimmed off the N-terminus (0) and C-terminus (L-1) ends of the peptide sequence. These symbols will be ignored
+        in the distance calculation.
+    fixed_gappos : bool
+        If True, insert gaps at a fixed position after the cysteine residue statring the CDR3 (typically position 6).
+        If False, find the "optimal" position for inserting the gaps to make up the difference in length
+    cutoff:
+        Will eleminate distances > cutoff to make efficient
+        use of sparse matrices.
+    n_jobs:
+        Number of jobs (processes) to use for the pairwise distance calculation
+    """
+
+    def __init__(
+        self,
+        dist_weight=3,
+        gap_penalty=4,
+        ntrim=3,
+        ctrim=2,
+        fixed_gappos=True,
+        cutoff: Union[None, int] = None,
+        n_jobs: Union[None, int] = None,
```

Comment:
I suggest making the arguments *keyword only*. This forces the user to be explicit and write more readable code. It also allows us to add/reorder parameters at a later point without breaking API compatibility. It's considered good practice for basically any function with many parameters which you don't expect the user to pass as positional arguments. 


```suggestion
        self,
        *,
        dist_weight=3,
        gap_penalty=4,
        ntrim=3,
        ctrim=2,
        fixed_gappos=True,
        cutoff: Union[None, int] = None,
        n_jobs: Union[None, int] = None,
```